nuxeo-ai-core/src/main/java/org/nuxeo/ai/functions/SaveEnrichmentFunction.java

```diff
@@ -41,30 +42,52 @@ public class SaveEnrichmentFunction extends AbstractEnrichmentConsumer {
     public void accept(EnrichmentMetadata metadata) {
         TransactionHelper.runInTransaction(() -> CoreInstance.doPrivileged(metadata.context.repositoryName, session -> {
             DocMetadataService docMetadataService = Framework.getService(DocMetadataService.class);
+
+            log.debug("Saving enrichment for document {}.", metadata.context.documentRef);
             DocumentModel doc = docMetadataService.saveEnrichment(session, metadata);
-            if (doc != null) {
-                try {
-                    if (!doc.isImmutable()) {
-                        log.debug("Saving enrichment for document {}.", doc.getId());
-                        session.saveDocument(doc);
-                    } else {
-                        log.error("Attempt to write into an Immutable Document Model id: {}, AI Model name {}",
-                                doc.getId(), metadata.getModelName());
-                    }
-                } catch (DocumentValidationException e) {
-                    log.warn("Failed to save document enrichment data for {}; error {}", metadata.context.documentRef,
-                            e.getMessage());
-                    if (log.isDebugEnabled()) {
-                        // log field violations
-                        List<ConstraintViolation> violations = e.getReport().asList();
-                        for (ConstraintViolation violation : violations) {
-                            log.debug("Constraint: {}, Value: {}", violation.getConstraint().getDescription(),
-                                    violation.getInvalidValue());
-                        }
+
+            if (doc == null) {
+                log.warn("Failed to save enrichment for document {}.", metadata.context.documentRef);
+                return null;
+            }
+
+            log.debug("Checking if the document is checked out and if a base version exists for the document {}.",
+                    doc.getId());
+
+            DocumentRef baseVersionRef = session.getBaseVersion(doc.getRef());
+
+            if (baseVersionRef == null && !doc.isCheckedOut()) {
+                log.warn(
+                        "Failed to save enrichment. The document is not checked out and no base version was found for document {}.",
```

Comment:
You could state that the document is corrupted and require a manual intervention to be fixed, this is not something to be expected, could be better at error level.

Reply:
This has been changed in [f0c7c09](https://github.com/nuxeo/nuxeo-ai/commit/f0c7c09a20df8eeb38f0a4efb2a42cae1a10abc6) - https://github.com/nuxeo/nuxeo-ai/compare/dd3b862658a35f863a8a0ab7b85739af3a58b189..f0c7c09a20df8eeb38f0a4efb2a42cae1a10abc6